release-notes.txt

```diff
@@ -1,3 +1,521 @@
+capabilities and comes with support for a range of devices that are typically
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release:
+===================
+
+The 2019.10 release includes:
+
+ - initial support for SUIT firmware updates
+ - USB CDC-ACM serial communication
+ - complete rewrite of TI CC110x radio driver
+ - initial support for IPv6 fragmentation
+ - DTLS support in the sock networking stack
+ - complete blockwise messaging for gcoap and nanocoap
+ - as always, bug fixes and documentation updates
+
+About 460 pull requests, composed of 950 commits, have been merged since the
+last release, and about 70 issues have been solved. 57 people contributed with
+code in 105 days. Approximately 2000 files have been touched with 129000
+insertions and 25000 deletions.
+
+
+Notations used below:
+=====================
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+New features and changes
+========================
+
+System libraries
+----------------
+    + sys/arduino: Added millis()
+    * sys/arduino: make ADC feature optional
+    + sys/fmt: Added submodule fmt_table for printing tables
+    + sys/log: add module for colorized logging
+    + sys/riotboot: add initial image digest verification
+    + sys/shell: add heap command
+    + sys/shell: cancel current line on CTRL-C.
+    * shell/sc_gnrc_ipv6_nib: list and edit authoritative border router (ABR) list
+    + sys/suit: initial support for SUIT firmware updates
+    * sys: add schedstatistics module to remove dependency on xtimer
+    * can: use memarray for pkt and router allocation
+    + sys: single interrupt handler thread for interrupts in modules with blocking
+      functions (but likely to be removed -- see Deprecations section below)
+
+Networking
+----------
+    + sys: Add (CoAP) Link Format module
+    + sys: add credman (D)TLS credential management module
+    + sys: new sock submodule for DTLS
+    + gnrc_ipv6_ext_frag: Initial import of IPv6 fragmentation and reassembly
+    + gnrc_netif: allow for wait of minimum time between sends
+    * gnrc_netif: add netif setter (part of refactoring to a pointer-based
+      approach)
+    * gnrc_sixlowpan_frag_vrb: add gnrc_sixlowpan_frag_vrb_from_route()
+    + sixlowpan_sfr: initial import Selective Fragment Recovery definitions
```

Comment:
The factoring out of the reassembly buffer in its own module (`gnrc_sixlowpan_frag_rb`, edit https://github.com/RIOT-OS/RIOT/pull/12318 and https://github.com/RIOT-OS/RIOT/pull/12324) on the other hand should be mentioned here.

Reply:
Fixed in d8aafc0